src/backend/catalog/namespace.c

```diff
@@ -1478,14 +1523,70 @@ MatchNamedCall(HeapTuple proctup, int nargs, List *argnames,
 	Assert(ap == nargs);		/* processed all actual parameters */
 
 	/* Check for default arguments */
+	*defaults = NIL;
 	if (nargs < pronargs)
 	{
 		int			first_arg_with_default = pronargs - procform->pronargdefaults;
-
+		Datum		proargdefaults;
+		char		*str;
+		List		*argdefaults = NIL;
+		bool		isnull;
+		bool		special = false;
+		ListCell	*def_item = NULL;
+
+		proargdefaults = SysCacheGetAttr(PROCOID, proctup,
+										 Anum_pg_proc_proargdefaults,
+										 &isnull);
+
+		if (!isnull)
+		{
+			str = TextDatumGetCString(proargdefaults);
+			argdefaults = castNode(List, stringToNode(str));
+			def_item = list_head(argdefaults);
+			special = IsA(lfirst(def_item), FuncDefault) ? true : false;
```

Comment:
nit: is there any better name than `special`?

Reply:
Yup, changed!